src/antidifferentiation.jl

```diff
@@ -0,0 +1,77 @@
+export antidifferentiate
+
+"""
+    antidifferentiate(p::AbstractPolynomialLike, v::AbstractVariable, deg::Union{Int, Val}=1)
+
+Antidifferentiate `deg` times the polynomial `p` by the variable `v`. The free constant involved by
+the antidifferentiation is set to 0.
+
+
+    antidifferentiate(p::AbstractPolynomialLike, vs, deg::Union{Int, Val}=1)
+
+Antidifferentiate `deg` times the polynomial `p` by the variables of the vector or
+tuple of variable `vs` and return an array of dimension `deg`. It is recommended
+to pass `deg` as a `Val` instance when the degree is known at compile time, e.g.
+`antidifferentiate(p, v, Val{2}())` instead of `antidifferentiate(p, x, 2)`, as this
+will help the compiler infer the return type.
+
+### Examples
+
+```julia
+p = 3x^2*y + x + 2y + 1
+antidifferentiate(p, x) # should return 3x^3* + 1/2*x + 2xy + x
+antidifferentiate(p, x, Val{1}()) # equivalent to the above
+antidifferentiate(p, (x, y)) # should return [3x^3* + 1/2*x + 2xy + x, 3/2x^2*y^2 + xy + y^2 + y]
+```
+"""
+function antidifferentiate end
+
+# Fallback for everything else
+antidifferentiate(α::T, v::AbstractVariable) where {T} = α * v
+antidifferentiate(v1::AbstractVariable, v2::AbstractVariable) = v1 == v2 ? 1 / 2 * v1 * v2 : v1 * v2
```

Comment:
I am a bit concerned about defaulting to `Float64`, this might also be used with polynomials on `Rational{BigInt}` for instance, I would use `Rational{Int}` as it is more neutral so it will preserve the type of the polynomial it is used with
```julia
julia> typeof(1.0 * (1 // 2))
Float64

julia> typeof((big(1)) * (1 // 2))
Rational{BigInt}

julia> typeof((big(1) // big(2)) * (1 // 2))
Rational{BigInt}
```

Reply:
you're right, I fixed it in my last commit